loadtest/control/gencontroller/controller.go

```diff
@@ -122,6 +123,21 @@ func (c *GenController) Run() {
 		}
 	}
 
+	// Wait for all teams to be created, so all users can join them all.
+	for st.get(StateTargetTeams) != c.config.NumTeams {
+		time.Sleep(time.Second)
+	}
+
+	// Join all teams
+	if resp := c.joinAllTeams(c.user); resp.Err != nil {
```

Comment:
This just made me realize. createTeam is just called once per user, because it's not repeated. So we can never create more teams than the number of users. I guess this doesn't come up in practice, but it's an important limitation to note.

Reply:
Yup, that's true. I'll add a config validation.

Reply:
Fixed in https://github.com/mattermost/mattermost-load-test-ng/pull/608/commits/81adf7665e33880042bdc56958cfdfa98af910e2